client/lib/url/decode-utils.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * Internal dependencies
+ */
+import { URL as TypedURL } from 'types';
+
+interface Stringable {
+	toString: () => string;
+}
+
+/**
+ * Wrap decodeURI in a try / catch block to prevent `URIError` on invalid input
+ * Passing a non-string value will return an empty string.
+ * @param  encodedURI URI to attempt to decode
+ * @return            Decoded URI (or passed in value on error)
+ */
+export function decodeURIIfValid( encodedURI: string | Stringable ): TypedURL {
+	let encodedURIString: string;
+
+	if ( encodedURI as Stringable ) {
+		encodedURIString = encodedURI.toString();
+	} else {
+		encodedURIString = encodedURI as string;
```

Comment:
With an opThis doesn't look right, `string` will be handled in the other branch. We'll be hitting the Falsy cases here: `null | undefined | false | 0 | ''`.